src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -63,18 +66,46 @@ def validate_location(namespace):
 
 # region VM Create Validators
 
-def _validate_vm_create_storage_profile(namespace):
+def _validate_vm_create_storage_profile(namespace, for_scale_set=False):  # pylint: disable=too-many-branches, too-many-statements
 
     from azure.cli.command_modules.vm._actions import load_images_from_aliases_doc
 
-    image = namespace.image
+    image = namespace.image or ''
 
-    # 1 - Create native disk with VHD URI
-    if image.lower().endswith('.vhd'):
-        namespace.storage_profile = StorageProfile.SACustomImage
+    if not for_scale_set:
+        if namespace.managed_os_disk:
+            if namespace.image:
+                raise CLIError("'--image' is not applicable when attach to an existing os disk")
+        else:
+            if not image:
+                raise CLIError("Please provide parameter value to '--image'")
+
+    if image.lower().endswith('.vhd') or namespace.managed_os_disk:
         if not namespace.os_type:
             raise CLIError('--os-type TYPE is required for a native OS VHD disk.')
-        return
+
+    valid_managed_skus = ['premium_lrs', 'standard_lrs']
+    valid_unmanaged_skus = valid_managed_skus + ['standard_grs', 'standard_ragrs', 'standard_zrs']
```

Comment:
I would leave in the **enum_choice_list(StorageSku) in params and then just do the special case validation for managed disk here.